clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -32,6 +118,10 @@
         this.rate = rate;
     }
 
+    public abstract DataOutputStream wrapForOutput(ByteBufferOutputStream buffer, byte messageVersion, int bufferSize);
+
+    public abstract DataInputStream wrapForInput(ByteBufferInputStream buffer, byte messageVersion);
```

Comment:
I see. I agree that making the inner class of `ByteBufferInputStream` top-level would be a little better. It would remove one level of indirection for the compressed case and remain the same for the uncompressed case.